mapie/calibration.py

```diff
@@ -0,0 +1,502 @@
+        """
+        Fit estimator will calibrate the predicted proabilities from the output
+        of a classifier.
+
+        Parameters
+        ----------
+        X : ArrayLike of shape (n_samples, n_features)
+            Training data.
+        y : ArrayLike of shape (n_samples,)
+            Training labels.
+        sample_weight : Optional[ArrayLike] of shape (n_samples,)
+            Sample weights for fitting the out-of-fold models.
+            If None, then samples are equally weighted.
+            If some weights are null,
+            their corresponding observations are removed
+            before the fitting process and hence have no residuals.
+            If weights are non-uniform, residuals are still uniformly weighted.
+            Note that the sample weight defined are only for the training, not
+            for the calibration procedure.
+            By default ``None``.
+        X_calib : Optional[ArrayLike] of shape (n_calib_samples, n_features)
+            Calibration data.
+        y_calib : Optional[ArrayLike] of shape (n_calib_samples,)
+            Calibration labels.
+        sample_weight_calib : Optional[ArrayLike] of shape (n_samples,)
+            Sample weights for calib for fitting the out-of-fold models.
+            If None, then samples are equally weighted.
+            If some weights are null,
+            their corresponding observations are removed
+            before the fitting process and hence have no residuals.
+            If weights are non-uniform, residuals are still uniformly weighted.
+            Note that the sample weight defined are only for the training, not
+            for the calibration procedure.
+            By default ``None``.
+        calib_size : Optional[float]
+            If X_calib and y_calib are not defined, then the calibration
+            dataset is created with the split defined by calib_size.
+        random_state : int, RandomState instance or None, default=None
+            For the ``sklearn.model_selection.train_test_split`` documentation.
+            Controls the shuffling applied to the data before applying the
+            split.
+            Pass an int for reproducible output across multiple function calls.
+            See :term:`Glossary <random_state>`.
+        shuffle : bool, default=True
+            For the ``sklearn.model_selection.train_test_split`` documentation.
+            Whether or not to shuffle the data before splitting.
+            If shuffle=False
+            then stratify must be None.
+        stratify : array-like, default=None
+            For the ``sklearn.model_selection.train_test_split`` documentation.
+            If not None, data is split in a stratified fashion, using this as
+            the class labels.
+            Read more in the :ref:`User Guide <stratification>`.
+
+        Returns
+        -------
+        MapieCalibrator
+            The model itself.
```

Comment:
I don't think you need to worry about the split between calibration and test data in this part of the code. I think it is up to the user to split its data (just like for the rest of MAPIE). Moreover, it will make the code much more readable

Reply:
done